src/neo/Wallets/Wallet.cs

```diff
@@ -527,22 +527,33 @@ public long CalculateNetworkFee(DataCache snapshot, Transaction tx)
             int size = Transaction.HeaderSize + tx.Signers.GetVarSize() + tx.Attributes.GetVarSize() + tx.Script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
             uint exec_fee_factor = NativeContract.Policy.GetExecFeeFactor(snapshot);
             long networkFee = 0;
+            int index = -1;
             foreach (UInt160 hash in hashes)
             {
+                index++;
                 byte[] witness_script = GetAccount(hash)?.Contract?.Script;
+                byte[] invocationScript = null;
 
-                if (witness_script is null && tx.Witnesses != null)
+                if (tx.Witnesses != null)
                 {
-                    // Try to find the script in the witnesses
-
-                    foreach (var witness in tx.Witnesses)
+                    if (witness_script is null)
                     {
-                        if (witness.ScriptHash == hash)
+                        // Try to find the script in the witnesses
+
+                        foreach (var witness in tx.Witnesses)
                         {
-                            witness_script = witness.VerificationScript;
-                            break;
+                            if (witness.ScriptHash == hash)
+                            {
+                                witness_script = witness.VerificationScript;
+                                invocationScript = witness.InvocationScript;
+                                break;
+                            }
                         }
                     }
+                    else
+                    {
+                        invocationScript = tx.Witnesses[index].InvocationScript;
+                    }
```

Comment:
```suggestion
                    if (witness_script is null)
                    {
                        // Then it's a contract-based witness, so try to get the corresponding invocation script for it
                        invocationScript = tx.Witnesses[index]?.InvocationScript;
                    }
```
I.e. if we weren't able to find a proper sig/multisig verification script among transaction witnesses and **`witness_script` is still null**, then the last possible option is that we're dealing with the contract witness. So trying to fill its invocation script.